web/source/osk/inlinedOskView.ts

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Function     _Unload
+     * Scope        Private
+     * Description  Clears OSK variables prior to exit (JMD 1.9.1 - relocation of local variables 3/9/10)
+     */
+    _Unload() {
+      this.keyboardView = null;
+      this.bannerView = null;
+      this._Box = null;
+    }
+
+    protected setBoxStyling() {
+      const s = this._Box.style;
+      s.display  = 'none';
+      // Positioned with no relative offset from its default position.
+      // This allows _Box to still serve as an offsetParent for keytip & subkey menu positioning.
+      s.position = 'relative';
+    }
+
+    protected postKeyboardLoad() {
+      this._Visible = false;  // I3363 (Build 301)
+
+      this._Box.onmouseover = this._VKbdMouseOver;
+      this._Box.onmouseout = this._VKbdMouseOut;
+
+      if(this.displayIfActive) {
+        this.present();
+      }
+    }
+
+    /**
+     * Function     restorePosition
+     * Scope        Public
+     * @param       {boolean?}      keepDefaultPosition  If true, does not reset the default x,y set by `setRect`.
+     *                                                   If false or omitted, resets the default x,y as well.
+     * Description  Move OSK back to default position, floating under active input element
+     */
+    ['restorePosition']: (keepDefaultPosition?: boolean) => void = function(this: AnchoredOSKView, keepDefaultPosition?: boolean) {
+      return;
+    }.bind(this);
+
+    /**
+     * Function     _VKbdMouseOver
+     * Scope        Private
+     * @param       {Object}      e      event
+     * Description  Activate the KMW UI on mouse over
+     */
+    private _VKbdMouseOver = function(this: AnchoredOSKView, e) {
+      com.keyman.singleton.uiManager.setActivatingUI(true);
+    }.bind(this);
+
+    /**
+     * Function     _VKbdMouseOut
+     * Scope        Private
+     * @param       {Object}      e      event
+     * Description  Cancel activation of KMW UI on mouse out
+     */
+    private _VKbdMouseOut = function(this: AnchoredOSKView, e) {
+      com.keyman.singleton.uiManager.setActivatingUI(false);
+    }.bind(this);
+
+    /**
+     * Get the wanted height of the OSK
+     *  @return   {number}    height in pixels
+     **/
+    getDefaultKeyboardHeight(): number {
+      if(this.keyboardView instanceof VisualKeyboard) {
+        return this.keyboardView.height;
+      } else {
+        // Should probably refine, but it's a decent stopgap.
+        return this.computedHeight;
+      }
+    }
+
+    /**
+     * Get the wanted width of the OSK
+     *
+     *  @return   {number}    height in pixels
+     **/
+    getDefaultWidth(): number {
+      return this.computedWidth;
+    }
+
+    /**
+     * Allow the UI or page to set the position and size of the OSK
+     * and (optionally) override user repositioning or sizing
+     *
+     * @param       {Object.<string,number>}   p  Array object with position and size of OSK container
+    **/
+    ['setRect'](p: OSKRect) {
```

Comment:
We use `['setRect']` style here, but `getDefaultWidth()` style above -- is there a reason for the inconsistency?

Reply:
`getDefaultWidth()` isn't existing public API.

`setRect()`, on the other hand, _is_ defined... even if not properly supported here yet.  (It's intended for use with desktop OSKs, now known as the `FloatingOSKView`.)  The OSK APIs will probably need a rework at some point to reflect what functionality each OSK type supports.